File: backend/bookingmovieticketapp/app/Models/ShowTime.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class ShowTime extends Model
{
    use HasFactory;
    protected $table = 'showtimes';
    protected $primaryKey = 'id';
    public $incrementing = true;
    public $timestamps = false;
    protected $fillable = [
        'movieid',
        'roomid',
        'showdate',
        'starttime',
        'price',
        'isactive',
    ];

    public function movie()
    {
        return $this->belongsTo(Movie::class, 'movieid');
    }

    public function room()
    {
        return $this->belongsTo(Room::class, 'roomid');
    }

    public function bookings()
    {
        return $this->hasMany(Booking::class, 'showtimeid');
    }
}

File: frontendapp/lib/models/showtime.dart
class Showtime {
  final int id;
  final int movieId;
  final int roomId;
  final DateTime showDate;
  final DateTime startTime;
  final int price;
  final bool isactive;

  Showtime({
    required this.id,
    required this.movieId,
    required this.roomId,
    required this.showDate,
    required this.startTime,
    required this.price,
    required this.isactive,
  });

  factory Showtime.fromJson(Map<String, dynamic> json) => Showtime(
    id: json['id'],
    movieId: json['movieId'],
    roomId: json['roomId'],
    showDate: DateTime.parse(json['showdate']),
    startTime: DateTime.parse(json['starttime']),
    price: json['price'],
    isactive: json['isactive'],
  );
} 
